content/references/rippled-api/api-conventions/basic-data-types.md

```diff
@@ -21,6 +21,33 @@ Each closed [Ledger](ledger-data-formats.html) has a [Ledger Index][] and a [Has
 {% include '_snippets/data_types/hash.md' %}
 <!--{#_ #}-->
 
+### Hash Prefixes
+[[Source]<br>](https://github.com/ripple/rippled/blob/master/src/ripple/protocol/HashPrefix.h "Source")
+
+In many cases, the XRP Ledger prefixes an object's binary data with a 4-byte code before calculating its hash, so that objects of different types have different hashes even if their binary formats are the same. The 4-byte codes are structured as three bytes of ASCII followed by a zero byte.
```

Comment:
>The 4-byte codes are structured as three bytes of ASCII followed by a zero byte.

Technically, this isn't required: the prefix is just four bytes long and, so far, we've used three ASCII characters and a zero byte, mostly because it simplfies debugging (you can just treat the thing as a string and dump it on the console or in a log file). But it could, just as well, be `0x4D454F57` (aka "MEOW").

Also note that "three bytes of ASCII" is a bit confusing. If you really want to say that, perhaps something like "current codes consist of 3 characters from the 26-character English alphabet (encoded as ASCII) followed by a 0 byte." You don't need to link to an ASCII table :smiley: